sql/core/src/main/scala/org/apache/spark/sql/jdbc/MySQLDialect.scala

```diff
@@ -259,6 +272,8 @@ private case class MySQLDialect() extends JdbcDialect with SQLConfHelper with No
     case StringType => Option(JdbcType("LONGTEXT", java.sql.Types.LONGVARCHAR))
     case ByteType => Option(JdbcType("TINYINT", java.sql.Types.TINYINT))
     case ShortType => Option(JdbcType("SMALLINT", java.sql.Types.SMALLINT))
+    // We override getJDBCType so that DoubleType is mapped to DOUBLE instead.
+    case DoubleType => Option(JdbcType("DOUBLE", java.sql.Types.DOUBLE))
```

Comment:
Is this change covered in the PR title or desc?

Reply:
Sorry, I added this later on when testing cast of types on MySQL 5.7. You are right, let me add this type to PR title and the issue description.

Reply:
I also added comment to this line, so we can understand why this change is necessary.